src/support/safe_math.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_safe_math_h
+#define wasm_safe_math_h
+
+namespace wasm {
+
+#if !__has_builtin(__builtin_add_overflow)
+template<typename T> bool __builtin_add_overflow(T a, T b, T* output) {
+  // Atm this only supports unsigned types.
+  static_assert(std::is_unsigned_v<T>);
```

Comment:
Why does this support only unsigned types and what do you mean by LLVM-isms? If LLVM's version supports both, is there a reason why we wouldn't use that?

Reply:
By LLVM-isms I mean the code uses other LLVM headers and various `llvm::` stuff so it isn't easy to extract the entire header. We could extract just the one function, I suppose, but it uses LLVM naming conventions (`AddOverflow`).

Detecting signed overflow is harder as it can't just be `a+b > a` as with unsigned (since the added value can be negative). I think it's just another comparison (looks like that's what LLVM does) but I didn't look into it.